File: app/src/main/java/data/datamodels/Articles.java
package data.datamodels;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.Index;
import android.arch.persistence.room.PrimaryKey;
import android.arch.persistence.room.TypeConverters;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import data.database.utils.NewsAppTypeConverter;

@Entity(tableName="articles",
        indices = {@Index(value = "description", unique = true)})

@TypeConverters(NewsAppTypeConverter.class)

public class Articles {

    @SerializedName("id")
    @Expose
    @PrimaryKey(autoGenerate = true)
    private int id;

    @SerializedName("author")
    @Expose
    private String author;
    @SerializedName("title")
    @Expose
    private String title;
    @SerializedName("description")
    @Expose
    private String description;
    @SerializedName("url")
    @Expose
    private String url;
    @SerializedName("urlToImage")
    @Expose
    private String urlToImage;

    @SerializedName("publishedAt")
    @Expose
    private String publishedAt;
    @SerializedName("content")
    @Expose
    private String content;

    @SerializedName("country")
    @Expose
    private String country;

    @SerializedName("toRead")
    @Expose
    private boolean toRead;

    @SerializedName("favorite")
    @Expose
    private boolean favorite;

    public boolean getFavorite() {
        return favorite;
    }

    public void setFavorite(boolean favorite) {
        this.favorite = favorite;
    }
    public boolean getToRead() {
        return toRead;
    }

    public void setToRead(boolean toRead) {
        this.toRead = toRead;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getPublishedAt() {
        return publishedAt;
    }

    public void setPublishedAt(String publishedAt) {
        this.publishedAt = publishedAt;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getUrlToImage() {
        return urlToImage;
    }
    public void setUrlToImage(String urlToImage) {
        this.urlToImage = urlToImage;
    }


}

File: app/src/main/java/com/example/newsapp/data/datamodels/Articles.kt
package com.example.newsapp.data.datamodels

import android.arch.persistence.room.Entity
import android.arch.persistence.room.Index
import android.arch.persistence.room.PrimaryKey
import android.arch.persistence.room.TypeConverters

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import com.example.newsapp.data.database.utils.NewsAppTypeConverter

@Entity(tableName = "articles", indices = [Index(value = ["description"], unique = true)])
@TypeConverters(NewsAppTypeConverter::class)
class Articles {

    @SerializedName("id")
    @Expose
    @PrimaryKey(autoGenerate = true)
    var id: Int = 0

    @SerializedName("author")
    @Expose
    var author: String? = null
    @SerializedName("title")
    @Expose
    var title: String? = null
    @SerializedName("description")
    @Expose
    var description: String? = null
    @SerializedName("url")
    @Expose
    var url: String? = null
    @SerializedName("urlToImage")
    @Expose
    var urlToImage: String? = null

    @SerializedName("publishedAt")
    @Expose
    var publishedAt: String? = null
    @SerializedName("content")
    @Expose
    var content: String? = null

    @SerializedName("country")
    @Expose
    var country: String? = null

    @SerializedName("toRead")
    @Expose
    var toRead: Boolean = false

    @SerializedName("favorite")
    @Expose
    var favorite: Boolean = false


}